### PR TITLE
[*] CORE : Redis cache manager

### PR DESCRIPTION
**Pull request by mdrbx**
# Redis Cache in Prestashop
## What ?

We want to implement a reliable way to use Redis Cache into Prestashop. There is a PECL package for PHP which is old enough now to be reliable in production.
## Why ?

Redis is faster, less restrictive, reliable and stable.
(see here for more technical infos : http://stackoverflow.com/questions/10558465/memcached-vs-redis)
## How ?

Include it aside of Memcached, APC and file system cache already included in Prestashop.
Like @firstred did on his own plugin, as it's done with Memcached, we should also implement multi-server redis feature. 
Localhost should also be specified by default.
# TODO
- New SQL DB for Multi server ( @xBorderie this is the point where we need help indications )
- Form in controllers/admin/AdminPerformanceController.php
- Multi server in classes/cache/CacheRedis.php > Following @firstred proposal
- Speficy and test localhost first

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/4510)

### CODE REVIEW

**Comment by firstred** [comment]
Nice one! Could you also try to add support for authentication and other databases? You can have a look at this module to make things easier: https://github.com/firstred/phprediscache
Maybe you can even copy the whole `CacheRedis.php` file

**Comment by Mooty13** [comment]
Interesting, I think our code are nearly the same. Adding auth could be great (as it's a new feature in Redis), but we should add a table like they did previously with memcache.
Sadly I don't know how we can manage to make this kind of improvements. Adding SQL to the installer is easy, but I don't think they want more tables in Prestashop for now.

We can add by default (for first configuration of redis cache I mean) local server to help newcomers on this. Lots of people still don't know about how redis works.

@Shudrum & @julienbourdeau how can we do this ? Can we add a table ?

**Comment by xBorderie** [comment]
Could you write more details about what you're doing, why you're doing it and how to use it? 
It would really help the code reviewer.

Thanks for your contribution!

**Comment by Mooty13** [comment]
I'll explain it in first post. I edit it right away.

File: classes/cache/CacheRedis.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	public function connect()
+	{
+        
+		if (class_exists('Redis') && extension_loaded('redis')){
+			$this->redis = new Redis();
+        } else {
+			return false;
+        }
+       
+        if (!defined('_PS_CACHE_REDIS_') || !_PS_CACHE_REDIS_ || !preg_match("/:/",_PS_CACHE_REDIS_)){
```
Comment (ZiZuu-store) [inline classes/cache/CacheRedis.php]: `strpos()` should suffice

**Comment by mickaelandrieu** [comment]
Sadly the performance page have been migrated to a new system, feel free to rebase your branch to the new system.

More now we have the Cache Component from Symfony which cache with a [Redis adapter](https://symfony.com/doc/3.4/components/cache/adapters/redis_adapter.html). As we are working on making all our caches manageable by this component, I think this is a duplicate.

Feel free to rebase and re-open this pull request if I'm wrong.

Thanks!

**Pull request closed without merging**